### PR TITLE
BODY_STYLE_EMPTY should produce same method name

### LINKED ISSUE #412: BODY_STYLE_BARE automagically changed into BODY_STYLE_EMPTY
I don't understand all the inner working of BODY_STYLE but ended having to hack around it:
The lines concerned are:
https://github.com/arskom/spyne/blob/master/spyne/decorator.py#L333-L339
which change a 'bare' body style into an 'empty' one.

I would be fine with it but this has implications on the name of the method:
https://github.com/arskom/spyne/blob/master/spyne/_base.py#L425-L429

So I ended up adding an optional dummy field on my in_message object so that spyne does not consider it as empty and does not change my method name. This is an unfortunate hack.

Maybe it should be:

    if body_style is BODY_STYLE_BARE or body_style is BODY_STYLE_EMPTY:
        self.name = in_message.Attributes.sub_name

or maybe Spyne should not try to guess that body_style == 'bare' can mean 'empty' and let the user specify it ?

### PR DESCRIPTION
**Pull request by xavfernandez**
as BODY_STYLE_BARE
closes #412

### CODE REVIEW

**Comment by plq** [comment]
I think we should test the empty case as well

**Comment by xavfernandez** [comment]
spyne tests seem to have an issue with pypy since some previous commits:
https://travis-ci.org/arskom/spyne/builds

**Comment by plq** [comment]
Yes, I'm yet to look at why

**Comment by xavfernandez** [comment]
Which empty case ? You meant the non-empty/normal case ?

**Comment by xavfernandez** [comment]
I would love to see it merged: could you specify the additional test you would like ?

**Comment by plq** [comment]
```
class SomeService(ServiceBase):
 @rpc(_body_style='bare', _returns=String)
 def some_call(ctx, request):
   return 'abc'
```

**Comment by xavfernandez** [comment]
Wanted to add an other test but now:
https://github.com/arskom/spyne/blob/master/spyne/application.py#L193-L196
is causing a problem.

File: spyne/test/test_service.py [hunk 1]
```diff
@@ -335,6 +337,44 @@ def some_call(ctx):
...
+        app = Application([SomeService], 'tns', in_protocol=Soap11(),
+                                                out_protocol=Soap11(cleanup_namespaces=True))
+        assert SomeService.public_methods['some_call'].name == 'some_call'
+
+        req = """
+        <soap11env:Envelope xmlns:soap11env="http://schemas.xmlsoap.org/soap/envelope/"
+                        xmlns:tns="tns">
+            <soap11env:Body>
+                <tns:some_call>
+                    <tns:EmptyRequest/>
```
Comment (xavfernandez) [inline spyne/test/test_service.py]: BODY_STYLE_EMPTY is not expecting that

**Comment by xavfernandez** [comment]
I think spyne should allow the user to specify _body_style='empty' and should not try to switch from 'bare' to 'empty'

**Comment by plq** [comment]
what kind of problem?

**Comment by xavfernandez** [comment]
Cf the test case test_soap_bare_empty_model_input_method_name:
when you post:

        <soap11env:Envelope xmlns:soap11env="http://schemas.xmlsoap.org/soap/envelope/"
                        xmlns:tns="tns">
            <soap11env:Body>
                <tns:some_call>
                    <tns:EmptyRequest/>
                </tns:some_call>
            </soap11env:Body>
        </soap11env:Envelope>

you get:

    TypeError: some_call() takes exactly 2 arguments (1 given)

because of:
https://github.com/arskom/spyne/blob/master/spyne/application.py#L193-L196

**Comment by plq** [comment]
no spyne is doing the right thing here, the request is empty so no argument is passed. you should remove the request argument

**Comment by xavfernandez** [comment]
Well if the user specify:

    @rpc(EmptyRequest, _body_style='bare', _returns=String)
    def some_call(ctx, request):

It means it is expecting an object EmptyRequest and if spyne receives it, I dont see why it should not pass it to the function.

If he did want an empty request, he would have written:

    @rpc(_body_style='bare', _returns=String)
    def some_call(ctx):

**Comment by plq** [comment]
Yes, but for bare mode that's not how it works. There's no difference between an empty explicit request and implicit one.

The bug here is that Spyne lets the server boot at all, not that empty bare request is not passed to the function.

**Comment by xavfernandez** [comment]
ok so the issue with the name isnt one either I guess ?

**Comment by xavfernandez** [comment]
Well I created #414 instead.

**Pull request closed without merging**